apps/server/src/shared/repo/school/school.repo.ts

```diff
@@ -24,9 +24,12 @@ export class SchoolRepo extends BaseDORepo<SchoolDO, School, ISchoolProperties>
 	}
 
 	async findBySchoolNumber(officialSchoolNumber: string): Promise<SchoolDO | null> {
-		const school: School | null = await this._em.findOne(School, { officialSchoolNumber });
+		const [schools, count] = await this._em.findAndCount(School, { officialSchoolNumber });
```

Comment:
the official school number should be unique per federal state
so, if possible the federal state could be added to the filter

Reply:
Thanks for the hint :)  It would be a bigger change because we havent got the federalstate in nest at the moment.
We shouldnt got problems with the current implementation because each federalstate has it own db or?

Reply:
indeed, THR, BRB, and NBC have different instances, but on DBC we have schools from everywhere

Reply:
Okay but then i would say that the schools from everywhere must have a unique officialSchoolNumber